### ISSUE TITLE
Could the code be simpler?

### ISSUE THREAD

**Issue by qbolec** [Bootstragram/zealit#1]
Hi! I found your project following your answer on stackoverflow to the problem, we also tried to solve.
The idea of using Proxy is quite cool.
But when reading the source code I have a feeling that it is a bit overcomplicated.
In parcticular, once you've gone past 
```
            if (v !== undefined) {
                return v
            }
```
we know that `v === undefined` so I do not really understand why do we have to check if `key` is `"toJSON"`. I believe that if `key==="toJSON"` then then `v` would not be `undefined` for most of the time, and if it would be `undefined`, then it would be better to know about it. 

Perhaps I'm confused here, so I would appreciate the example in which we need such a complicated thing, and not something super simple like `if(key in target) return target[key]; else throw ...` ?

**Comment by zertyuio** [comment]
Hi,

thanks for your interest in zealit, first issue, wow :)
Sorry for the delay, I was on holiday 🌴🏖🚴.

There are two objectives to this code:

1. detecting when a property is nonexistent
1. not being too restrictive for the developer

### Objective 1: Detecting when a property is nonexistent

Nonexistent doesn't always mean `undefined`.
For example, when you set a property to `undefined` inside an object, its value is `undefined`, but the property exists :

    const obj = zealit({ a: undefined })
    obj.a // no Error thrown
    obj.b // Error thrown

That's the reason why there is `if (fHasOwnProperty.call(target, key))` after `if (v !== undefined)`.

We could simplify and use the check `fHasOwnProperty` only, but most of the time, the value will not be `undefined`, so we try to preserve performance.

And `if (key in target)` is a good alternative to `fHasOwnProperty`, but has also a performance cost compared to `(v !== undefined)`.

### Objective 2: Not being too restrictive for the developer

That's the reason why there is `option.ignore.includes(key)`.
In fact, if you work with expressjs, there is many pieces of code like that:

```
function foo(bar) {
    if (bar.toJSON) {
        bar = bar.toJSON()
    }

    ...
}
```

If `bar` is a zealed object, an `Error` will be thrown.
But most of the time, the code is correct and we don't want to throw an `Error` in this case.
That's why there is the `ignore` option.

So now, if you want to know when `toJSON` is `undefined`, what you can do is to configure zealit like that:

```
zealit.option.ignore = []
// or
zealit.option.ignore = zealit.option.ignore.filter((k) => k !== 'toJSON')
```

And if you want to know when a property exists as `undefined`, we can add an option like that:

```
const obj = zealit({ a: undefined }, { strict: true })
obj.a // Error thrown
obj.b // Error thrown
```

What do you think ?

**Comment by zertyuio** [comment]
Hi,

We’ve just released the 2.4.0 update, with the `strict` option, in order to throw Error if the value is `undefined`, even if the property exists on the object.

Hope this helps.
